### PR TITLE
Multiple OnScrollListeners functionality.

### PR DESCRIPTION
**Pull request by tudorgk**
Deprecated `setOnScrollListener` and replaced it with `addOnScrollListener` and `removeOnScrollListener`.

### CODE REVIEW

File: demo/src/main/java/com/jude/dome/loadmore/RefreshAndMoreActivity.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import android.support.v4.widget.SwipeRefreshLayout;
 import android.support.v7.app.ActionBarActivity;
 import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
```
Comment (cpalasanu) [inline demo/src/main/java/com/jude/dome/loadmore/RefreshAndMoreActivity.java]: why was this added?

File: easyrecyclerview/src/main/java/com/jude/easyrecyclerview/EasyRecyclerView.java [hunk 2]
```diff
@@ -380,11 +386,44 @@ public void setRefreshingColor(int... col) {
      *
      * @param listener
      */
+
+    /**
+     * Set scroll listener to the recycler
+     *
+     * @deprecated Use {@link #addOnScrollListener(RecyclerView.OnScrollListener)} instead.
+     */
+    @Deprecated
```
Comment (cpalasanu) [inline easyrecyclerview/src/main/java/com/jude/easyrecyclerview/EasyRecyclerView.java:394]: maybe you can just add the listener to the list here and get rid of `mExternalOnScrollListener`?

Comment (tudorgk) [reply easyrecyclerview/src/main/java/com/jude/easyrecyclerview/EasyRecyclerView.java:394]: In these types of cases, we shouldn't remove functionality. It might break other people's projects if they are directly accessing the `mExternalOnScrollListener`.
